src/app/integrations/edit-page/step-configure/data-mapper/data-mapper-host.component.ts

```diff
@@ -129,45 +129,30 @@ export class DataMapperHostComponent extends FlowPage
   }
 
   createDocumentDefinition(dataShape: DataShape, isSource: boolean = false) {
-    if (!dataShape || !dataShape.type || !dataShape.kind) {
+    if (!dataShape || !dataShape.kind) {
       // skip
       return;
     }
     const type = dataShape.type;
     const kind = dataShape.kind;
-    // avoid passing duplicates off to the data mapper
```

Comment:
I don't know the background though, this duplicate check is no longer required?

Reply:
I don't think so, previously there were issues where the component was getting initialized more than once, those should be sorted.